### ISSUE TITLE
Possible stereoscopic frame alternation optimization similar to Oculus Link

### ISSUE THREAD

**Issue by xieve** [alvr-org/ALVR#584]
I've tried Oculus Link with USB 2.0 (the 3.0 cable is still on its way) and from what I've seen, it looks like they use a 144Hz stream in 72Hz mode or a 180Hz stream in 90Hz mode, and alternate left and right eye images. This uses more effectively the merits of modern-day compression, as the mostly miniscule difference between the two eyes potentially means a lot less data to transfer. I'm basing this assumption upon the fact that when the connection lags or breaks down, and you move your head, the visual artifacts are always around the area where there's a difference between both eyes. Furthermore, when there's framedrops or an unstable connection, sometimes the eyes swap for a frame. This is a potential downside, but it only occurs on bad connections and might be mitigated. If I understand correctly, this project currently streams both eyes next to each other, but please correct me if I'm wrong.

I'd be willing to submit a PR, but I frankly don't really know where to start. I have zero experiences in any of the APIs, and basic knowledge of C++, Rust and GLSL. Any pointers in which file to look are welcome.

TL;DR: Oculus Link alternates left and right eye in a 180Hz stream to make use of the codecs, and this might be viable for ALVR too.

**Comment by zarik5** [comment]
The codec h264 has an extension for stereoscopic video streams, that leverages the similarity of the two eye views, but it is not widely adopted and AFAIK not present in hardware acceleration APIs. Instead encoding alternated eye views on the same encoder will degrade considerably the performance, because at every frame you need to encode the difference between the eyes instead of just between the single eye views and the previous ones. So especially at high framerates alternating the eye views will be much worse for compression/bandwidth.
